ansible/roles/nginx/templates/nginx.conf.j2

```diff
@@ -59,6 +66,12 @@ http {
     }
 
     proxy_set_header X-OW-EXTRA-LOGGING $extra_logging;
+    # Set the request id generated by nginx as tid-header to the upstream.
+    # This tid is either the request-id generated by ngix or a tid, sent by the caller.
+    proxy_set_header OW-TID $ow_tid;
+
+    # Send the tid always back as header.
+    add_header OW-TID $ow_tid always;
```

Comment:
As  @ddragosd  mentioned this will be changed to `X-Request-Id` (per https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Common_non-standard_request_fields it's a quasi standard).

The header that's picked in the controller will be configurable, so you can use whatever front-door generates these ids for you without relying on this too much.